tests/python/pants_test/task_test_base.py

```diff
@@ -380,8 +382,12 @@ def invoke_tasks(self, target_closure=None, **context_kwargs):
       current_task_instance
     ] + run_after_task_instances
 
-    for tsk in all_task_instances:
-      tsk.execute()
+    try:
+      for tsk in all_task_instances:
+        tsk.execute()
+    except Exception as e:
+      setattr(e, '_context', context)
```

Comment:
This is super hacky. Consider circling back in a follow up before reliance on this gets out of control and either
1. splitting the method into two parts, one that returns context, run_before_synthesized_task_types, run_after_synthesized_task_types and one that consumes these 3 and actually invokes tasks
2. Always return a self.TaskInvocationResult but teach that type to say success/error or make it an enum or what have you.

Reply:
I would normally do (1), but since we already have the special `self.TaskInvocationResult` class we might as well use it to actually represent all types of results from invoking tasks, so will likely do (2) in a followup. Thank you for outlining how to effectively address this!